services/wallet/walletconnect/database.go

```diff
@@ -151,46 +184,66 @@ func GetSessionsByPairingTopic(db *sql.DB, pairingTopic Topic) ([]DbSession, err
 }
 
 // GetActiveSessions returns all active sessions (active and not expired) that have an expiry timestamp newer or equal to the given timestamp.
-func GetActiveSessions(db *sql.DB, expiryNotOlderThanTimestamp int64) ([]DbSession, error) {
-	querySQL := `
-	SELECT *
-	FROM
-		wallet_connect_sessions
-	WHERE
-		active != 0 AND
-		expiry >= ?
-	ORDER BY
-		expiry DESC`
+func GetActiveSessions(db *sql.DB, expiryNotOlderThanTimestamp int64) ([]DBSession, error) {
+	querySQL := selectAndJoinQueryStr + `
+		WHERE
+			sessions.active != 0 AND
+			sessions.expiry >= ?
+		ORDER BY
+			sessions.expiry DESC`
 
 	rows, err := db.Query(querySQL, expiryNotOlderThanTimestamp)
 	if err != nil {
 		return nil, err
 	}
 	defer rows.Close()
+	return scanSessions(rows)
+}
+
+// GetActiveSessions returns all sessions in the ascending order of creation time
+func GetSessions(db *sql.DB) ([]DBSession, error) {
```

Comment:
There should be a testnet/mainnet filter here 🤔

Reply:
Indeed, thanks for the heads up.

Reply:
Sorry for the confusion, I added the testnet filter only to the public API (`GetWalletConnectDapps`) which is returning the activity API and left improvements for non-public APIs for when exposed. I also added a test with the last amend here https://github.com/status-im/status-go/blob/24fbb2ff8fac955183a914348d50ab1fc3a62438/services/wallet/walletconnect/database_test.go#L292